src/worker.cc

```diff
@@ -806,10 +806,17 @@ void from_json(rapidjson::Document& doc, Options::Action action, Api& api) {
       options.set_shape_format(polyline5);
     } else if (*shape_format == "geojson") {
       options.set_shape_format(geojson);
+    } else if (*shape_format == "none") {
+      if (action == Options::height) {
+        throw valhalla_exception_t{164};
+      }
```

Comment:
its not clear to me why we need this. the height action will never look at `options.shape_format()` right? i think we shouldnt throw errors for superfluous parameters. or what am i missing?

Reply:
> the height action will never look at options.shape_format() right?

yes it does actually, to return either polyline5 or 6. in fact it's the only documented use of `shape_format` :)

Reply:
not to return though right! its the input format in that case hahaha, i had forgotten about that!

Reply:
we probably have to block this then for mapmatching too right?

Reply:
i shouldnt say i didnt check this but we should have a look quick while we are here

Reply:
https://github.com/valhalla/valhalla/blob/253443ed2b3c7838a85be0d2b89e1876a7d542d4/src/worker.cc#L839-L852

Reply:
looks like mapmatching forces to polyline6. kind of dumb, we should allow it but its a problem for another day

Reply:
for height as input, right.. for map matching not relevant, it's hard-coded to polyline6 it seems in worker.cc